pkg/plugin/vector/store/vectorapi.go

```diff
@@ -97,9 +131,15 @@ func (g *grafanaVectorAPI) Health(ctx context.Context) error {
 	return nil
 }
 
-func newGrafanaVectorAPI(s grafanaVectorAPISettings, secrets map[string]string) ReadVectorStore {
+func newGrafanaVectorAPI(s grafanaVectorAPISettings, authType string, authS AuthSettings, secrets map[string]string) (ReadVectorStore, error) {
 	return &grafanaVectorAPI{
-		client: &http.Client{},
-		url:    s.URL,
-	}
+		client:   &http.Client{},
+		url:      s.URL,
+		authType: authType,
+		authSettings: grafanaVectorAPIAuthSettings{
+			BasicAuthUser:     authS.BasicAuthUser,
+			BasicAuthPassword: secrets["basicAuthPassword"],
+			AuthHeaderValue:   secrets["authHeaderValue"],
```

Comment:
We should factor these secret keys out into constants and scope them (e.g. `grafanaVectorAPIBasicAuthPassword`?) since secrets are global and we may have some other section that needs to add a basic auth password later.